modules/REST/V1/Settings.php

```diff
@@ -107,6 +107,12 @@ public function update_item( $request ) {
 
 			if ( $result->has_errors() ) {
 				$errors->merge_from( $result );
+				continue;
+			}
+
+			$options_with_flush = array( 'rewrite', 'force_lang', 'hide_default' );
+			if ( in_array( $option_name, $options_with_flush, true ) ) {
+				flush_rewrite_rules();
 			}
```

Comment:
Maybe we should do this only if the value change?

```php
$prev_value = $this->options->get( $option_name );
$result     = $this->options->set( $option_name, $new_value );

if ( $result->has_errors() ) {
	$errors->merge_from( $result );
	continue;
}

if ( $this->options->get( $option_name ) === $prev_value ) {
	continue;
}

$options_with_flush = array( 'rewrite', 'force_lang', 'hide_default' );
if ( in_array( $option_name, $options_with_flush, true ) ) {
	flush_rewrite_rules();
}
```

What do you think?

Reply:
Why not, but I we got that way I propose to do nothing at all in such case.
See 84c2a709.